workspaces/adoption-insights/plugins/adoption-insights/src/api/index.ts

```diff
@@ -132,6 +172,16 @@ export class AdoptionInsightsApiClient implements AdoptionInsightsApi {
 
     const response = await this.fetchApi.fetch(url);
 
+    if (!response.body) {
+      throw new Error('Something went wrong.'); // NOSONAR
+    }
```

Comment:
Nits:

1. Can you update to have the similar error message format like you have in `(!response.ok)` block ?
2. Also it would be better to re-order there error blocks, i.e  first `(!response.ok)` and second check would be for (!response.body).

```suggestion
    if (!response.body) {
      throw new Error(`failed to fetch templates, status ${response.status}: Something went wrong.'`);  
    }
```